applications/games_menu/modules/menu.c

```diff
@@ -199,17 +328,74 @@ ImageDimensionStruct *GetImageDimension(const char *image_file)
 
 	if (strcasecmp(image_type, ".kmg") == 0)
 	{
+		/* Open the file */
 		file_t image_handle = fs_open(image_file, O_RDONLY);
+		if (image_handle == FILEHND_INVALID) {
+			return NULL;
+		}
 
-		if (image_handle == FILEHND_INVALID)
+		kmg_header_t hdr;
+		/* Read the header */
+		if (fs_read(image_handle, &hdr, sizeof(hdr)) != sizeof(hdr)) {
+			fs_close(image_handle);
+			return NULL;
+		}
+
+		/* Verify a few things */
+		if (hdr.magic != KMG_MAGIC || hdr.version != KMG_VERSION ||
+			hdr.platform != KMG_PLAT_DC)
+		{
+			fs_close(image_handle);
 			return NULL;
+		}
+		fs_close(image_handle);
```

Comment:
Why you not read image data from this file?

Reply:
This method is only to validate if the image is really correct, it is not necessary to read the complete information yet. For example, here read only the header.

Reply:
or I didn't understand?

Reply:
Got it. It's just that extra file operations slow things down. You need to touch the file system as little as possible.

Reply:
Yes, I do it because they can put corrupt files or simply files with changed extensions, let's say it is a validation to avoid human error.
This way I don't stop the menu operation, otherwise it stops.
After that, it saves the correct value in memory to avoid passing it again.
That's why I want to leave her.

Reply:
It is prepared to avoid this type of inconvenience.

Reply:
Okay, I'll test it soon and merge.

Reply:
Don't forget push last changes from another comments.

Reply:
No no, wait, :D!! In the Discord chat I left you a problem that I'm having with the emulator, I want to know who is causing it.